### PR TITLE
Background audio: Stop playback of audio on removal 

### LINKED ISSUE #11337: Background audio: Stop playback of audio on removal
## Bug Description

If you are playing background audio, if you click the remove button on the audio, the audio keeps playing for a 1 second. 

## Expected Behaviour

On removal of background audio, the audio should automatically stop. 

## Steps to Reproduce

1. Create a new story. 
2. Add background audio to a page. 
3. Start playing background audio. 
4. Remove background audio. 
5. Hear audio still playing. 

## Additional Context

- Plugin Version:
- WordPress Version:
- Operating System:
- Browser:

### PR DESCRIPTION
**Pull request by timarney**
## Context

If you are playing background audio, if you click the remove button on the audio, the audio keeps playing for a 1 second.

## Summary

Moves player ref up to parent component to allow the parent to stop the auto player first before updating state etc...

On removal of background audio, the audio should stop immediately.

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Create a new story.
2. Add background audio to a page.
3. Start playing background audio.
4. Remove background audio.
5. Hear audio stop playing immediately

### Does this PR have a security-related impact?

No

### Does this PR change what data or activity we track or use?

No

### Does this PR have a legal-related impact?

No

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #11337

### CODE REVIEW

**Comment by timarney** [comment]
@spacedmonkey can you give this one a test?  I tried throttling my network etc... but I can't seem to recreate the original issue.

**Review by spacedmonkey — APPROVED** [review]

File: packages/story-editor/src/components/panels/shared/media/backgroundAudioPanelContent.js [hunk 1]
```diff
@@ -298,7 +305,13 @@ function BackgroundAudioPanelContent({
...
             isExternal={!resource.id}
             options={options}
-            onRemove={() => updateBackgroundAudio(null)}
+            onRemove={() => {
+              if (playerRef.current) {
+                playerRef.current.pause();
+              }
+
+              updateBackgroundAudio(null);
+            }}
```
Comment (spacedmonkey) [inline packages/story-editor/src/components/panels/shared/media/backgroundAudioPanelContent.js]: Should this be wrapped in a `useCallback`?

**Comment by spacedmonkey** [comment]
Approved with one question?

File: packages/story-editor/src/components/panels/shared/media/backgroundAudioPanelContent.js (hunk 1 above)
Comment (timarney) [reply packages/story-editor/src/components/panels/shared/media/backgroundAudioPanelContent.js]: Makes sense will update

File: packages/story-editor/src/components/panels/shared/media/backgroundAudioPanelContent.js [hunk 2]
```diff
@@ -239,6 +246,13 @@ function BackgroundAudioPanelContent({
     []
   );
 
+  const handleRemove = useCallback(() => {
+    if (playerRef.current) {
+      playerRef.current.pause();
+    }
+    updateBackgroundAudio(null);
+  }, []);
```
Comment (spacedmonkey) [inline packages/story-editor/src/components/panels/shared/media/backgroundAudioPanelContent.js]:
```suggestion
  }, [updateBackgroundAudio, playerRef]);
```

**Review by swissspidy — APPROVED** [review]
Really having troubles reproducing the original issue, but change seems reasonable

**Comment by timarney** [comment]
@swissspidy I think this one is ready to merge.

https://github.com/GoogleForCreators/web-stories-wp/issues/11337#issuecomment-1166275575

**Pull request merged**